sound/soc/sof/intel/hda-dai.c

```diff
@@ -159,6 +151,7 @@ static int hda_link_dma_cleanup(struct snd_pcm_substream *substream,
 	}
 	snd_soc_dai_set_dma_data(cpu_dai, substream, NULL);
 	snd_hdac_ext_stream_release(hext_stream, HDAC_EXT_STREAM_TYPE_LINK);
+	snd_hdac_ext_stream_reset(hext_stream);
```

Comment:
somehow one of my main comments on this patch was removed or not saved, I think this patch should be split in code moves and changes. I am not able to visualize if this change is correct for example. This is a helper used in 3 different parts of the code, this looks scary when mixed with all kinds of changes.